src/postamble.js

```diff
@@ -189,6 +192,9 @@ function callMain(args) {
     return handleException(e);
 #endif // !PROXY_TO_PTHREAD
   } finally {
+#if ASSERTIONS
+    userCodeEntriesOnStack -= 1;
+#endif
```

Comment:
This special instrumentation of `main` worries me. What about during `emscripten_main_loop` callbacks? I think we should warn inside them as well.

The more I think about it, the more I think instrumenting the exports is the right thing, if we want a warning for this.

Reply:
`emscripten_main_loop` and all the other places we re-enter already use `callUserCallback`, so they are handled already.

In fact it would be really good to move main itself to be run via `callUserback` to avoid this kind of duplication of logic.

Reply:
I see now, thanks.

Ok, then the direction in this PR sgtm. Maybe calling main via `callUserCallback` is nice, but if it increases code size, then since `main` is really special I'm not worried about that.

In this case, my main open question here is my other comment about the terminology of "user".